### ISSUE TITLE
support microprofile JWT 1.1

### ISSUE THREAD

**Issue by chunlongliang-ibm** [OpenLiberty/open-liberty#4479]
This feature implements MP-JWT 1.1 spec

https://github.com/eclipse/microprofile-jwt-auth/blob/master/spec/src/main/asciidoc/configuration.asciidoc

**Comment by gkwan-ibm** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval (note that this is a new approval step -- comments welcome on how to improve it):

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the new serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Queues -- Which L2 / L3 queues will handle PMRs for this feature? Do the respective L2 / L3 teams know they are supporting it?

**Comment by dave-waddling** [comment]
**FAT Focal Approval**: Please can you complete the Feature Test Summary issue (#4667  that has been added to your Epic. This issue is what is used to review the FAT testing and obtain the FAT focal approval. Many thanks.

**Comment by chunlongliang-ibm** [comment]
Request for serviceability approval:
@gkwan-ibm 
WAD-- identified major problem user may see while using this features. 
Test and Demo -- A live demo and feature overview were given to both L2 and L3, all questions have been answered.
SVT -- Rumana performs persona test, does not find new serviceability issue

Queues -- Retain PMR and APAR: handled by WL3WSS.Salesforce Q name:  WAS L3: Security SSO

**Comment by chunlongliang-ibm** [comment]
@jhanders34 , For performance approval, There is no performance impact. The main change is to accept configuration property from config source per mpConfig-1.x features

**Comment by rumanaHaque** [comment]
Completed SVT testing for this Feature, with single request and 24 hr single server stress test.
Here if the link to the test -
https://github.ibm.com/was-liberty/WS-CD-Open/issues/12930

**Comment by bsbyrd1** [comment]
My approval is based on the sizing of the ID work, and the expectation that the ID work can be completed before the last ID translation drop.

**Comment by dave-waddling** [comment]
**FAT Focal Approval:** Hi, the WAD describes the addition of an MP JWT TCK into our automated testing however based on the Feature Test Summary it looks like the TCK won't be automated in time. I've spoken to @garypicher  and @brutif  about this and the outcome is that I'm not going to be able to grant a FAT Focal Approval at this time.

**Comment by brutif** [comment]
#4787 tracks getting the initial tck code into github in quarantine mode while tck issues are worked through.

**Comment by chunlongliang-ibm** [comment]
[STE_for_MP_JWT.pptx](https://github.com/OpenLiberty/open-liberty/files/2311347/STE_for_MP_JWT.pptx)

**Comment by marikaj123** [comment]
Added STE approval on behalf of Swati.

**Comment by marikaj123** [comment]
Blog GA Info: https://github.ibm.com/was-liberty/WS-CD-Open/issues/13154

**Comment by ayoho** [comment]
@chunlongliang-ibm and @brutif: Is there any other work needing to be done for this feature that should be tracked under this item?

**Comment by brutif** [comment]
holding open pending tck updates from tck owner and getting the fat bucket out of quarantine

**Comment by atosak** [comment]
@brutif , is there an update of the TCK work?

**Comment by brutif** [comment]
@atosak  tck updates are going in on pull #4856 and will hopefully be completed in the next few days

**Comment by ayoho** [comment]
Closing now that #4856 has been delivered.
